exporter/splunkhecexporter/exporter_test.go

```diff
@@ -201,23 +202,28 @@ func TestConsumeMetricsData(t *testing.T) {
 			md:   generateLargeBatch(),
 			reqTestFunc: func(t *testing.T, r *http.Request) {
 				body, err := io.ReadAll(r.Body)
-				if err != nil {
-					t.Fatal(err)
-				}
+				assert.NoError(t, err)
 				assert.Equal(t, "keep-alive", r.Header.Get("Connection"))
 				assert.Equal(t, "application/json", r.Header.Get("Content-Type"))
 				assert.Equal(t, "OpenTelemetry-Collector Splunk Exporter/v0.0.1", r.Header.Get("User-Agent"))
 				assert.Equal(t, "Splunk 1234", r.Header.Get("Authorization"))
-				assert.Equal(t, "gzip", r.Header.Get("Content-Encoding"))
-				zipReader, err := gzip.NewReader(bytes.NewReader(body))
-				assert.NoError(t, err)
-				bodyBytes, _ := io.ReadAll(zipReader)
-				firstPayload := strings.Split(string(bodyBytes), "}{")[0]
-				var metric splunk.Event
-				err = json.Unmarshal([]byte(firstPayload+"}"), &metric)
-				if err != nil {
-					t.Fatal(err)
+				bodyBytes := body
+				// the last batch might not be zipped.
+				if "gzip" == r.Header.Get("Content-Encoding") {
+					zipReader, err := gzip.NewReader(bytes.NewReader(body))
+					require.NoError(t, err)
+					bodyBytes, _ = io.ReadAll(zipReader)
+				}
+
+				events := strings.Split(string(bodyBytes), "}{")
+				firstPayload := events[0]
```

Comment:
also sometimes the last batch had just one event. We fix it here by doing a better job separating HEC events.